platforms/cuda/src/CudaTorchKernels.cpp

```diff
@@ -94,13 +111,16 @@ double CudaCalcTorchForceKernel::execute(ContextImpl& context, bool includeForce
 
     // Copy the atomic positions and simulation box to PyTorch tensors
     {
-        ContextSelector selector(cu);
+        ContextSelector selector(cu); // Switch to the OpenMM context
         void* inputArgs[] = {&posData, &boxData, &cu.getPosq().getDevicePointer(), &cu.getAtomIndexArray().getDevicePointer(),
                 &numParticles, cu.getPeriodicBoxVecXPointer(), cu.getPeriodicBoxVecYPointer(), cu.getPeriodicBoxVecZPointer()};
         cu.executeKernel(copyInputsKernel, inputArgs, numParticles);
-        CHECK_RESULT(cuCtxSynchronize(), "Error synchronizing CUDA context"); // Synchronize before switching to the PyTorch context
+        CHECK_RESULT(cuCtxSynchronize(), "Failed to synchronize the CUDA context"); // Synchronize before switching to the PyTorch context
     }
 
+    // Push to the PyTorch context
+    CHECK_RESULT(cuCtxPushCurrent(primaryContext), "Failed to push the CUDA context");
```

Comment:
If we want to be completely safe, we probably should move this up to the top of `execute()`.  Most likely it's ok to call `data_ptr()` on a tensor without setting the context first, but it seems good to be safe.

Reply:
If you can fix this last detail, we can get this merged and release the OpenMM 8 beta.

Reply:
Done https://github.com/openmm/openmm-torch/pull/80/commits/beacb07aa6607a12d980920d7689a5c9b960c686!